### PR TITLE
Minor refactor ( add BoundsIntersect function )

### PR DESCRIPTION
**Pull request by justgo97**
Inspired by https://github.com/DrBeef/Lambda1VR/blob/master/Projects/Android/jni/src/Xash3D/xash3d/engine/server/sv_world.c

### CODE REVIEW

File: rehlds/engine/world.cpp [hunk 1]
```diff
@@ -319,6 +319,15 @@ void SV_UnlinkEdict(edict_t *ent)
 	ent->area.prev = ent->area.next = nullptr;
 }
 
+qboolean SV_BoundsIntersect(const vec3_t mins1, const vec3_t maxs1, const vec3_t mins2, const vec3_t maxs2)
```
Comment (s1lentq) [inline rehlds/engine/world.cpp:322]: I would prefer to move this function to [engine/mathlib.cpp](https://github.com/dreamstalker/rehlds/blob/5ec8c29185a960ae31b090de735f8399a16b3042/rehlds/engine/mathlib.cpp#L419)

**Comment by s1lentq** [comment]
https://github.com/dreamstalker/rehlds/blob/5ec8c29185a960ae31b090de735f8399a16b3042/rehlds/engine/sv_user.cpp#L586-L599

You could use it here too like this

```
	fmin = check->v.absmin;
	fmax = check->v.absmax;

	if (check->v.flags & FL_CLIENT)
	{
		// trying to get interpolated values
		SV_GetTrueMinMax(e - 1, &fmin, &fmax);
	}

	if (!BoundsIntersect(pmove_mins, pmove_maxs, fmin, fmax))
		continue;
```

File: rehlds/engine/world.cpp (hunk 1 above)
Comment (justgo97) [reply rehlds/engine/world.cpp]: done

**Comment by justgo97** [comment]
> https://github.com/dreamstalker/rehlds/blob/5ec8c29185a960ae31b090de735f8399a16b3042/rehlds/engine/sv_user.cpp#L586-L599
> 
> You could use it here too like this
> 
> ```
> 	fmin = check->v.absmin;
> 	fmax = check->v.absmax;
> 
> 	if (check->v.flags & FL_CLIENT)
> 	{
> 		// trying to get interpolated values
> 		SV_GetTrueMinMax(e - 1, &fmin, &fmax);
> 	}
> 
> 	if (!BoundsIntersect(pmove_mins, pmove_maxs, fmin, fmax))
> 		continue;
> ```

Done, Nice catch

**Review by s1lentq — APPROVED** [review]

**Pull request merged**